### PR TITLE
Handle decoding panic

### PR DESCRIPTION
**Pull request by AgeManning**
## Description

It is possible that a non-utf8 string that is attempted to be decoded as an ENR can panic with the current decoding. This PR handles this edge case.

This also corrects `ed25519` doc tests to only be run with the feature flag.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -745,8 +748,10 @@ impl<K: EnrKey> FromStr for Enr<K> {
         }
         // support both enr prefix and not
         let mut decode_string = base64_string;
-        if &base64_string[..4] == "enr:" {
-            decode_string = &decode_string[4..];
+        if base64_string.starts_with("enr:") {
+            decode_string = decode_string
+                .get(4..)
+                .ok_or_else(|| "Invalid ENR string".to_string())?;
```
Comment (ordian) [inline src/lib.rs:754]: the `else` branch of `ok_or_else` is unreachable because of the `starts_with` check above
